<launch>
    <!-- Set to 1 if you want to run the map_server -->
    <arg name="map_server" default="1"/>
    <arg name="map_file" default="$(find mushr_mujoco_ros)/maps/empty.yaml" />
    <arg name="environment" default="block" />

    <!-- Could be racecar-mit, racecar-uw-tx2, or racecar-uw-nano -->
    <arg name="racecar_version" default="racecar-uw-nano" />

    <include file="$(find mushr_sim)/launch/keyboard_teleop.launch.xml"/>

    <node pkg="topic_tools" type="relay" name="keyboard_teleop_relay" args="/mux/ackermann_cmd_mux/input/teleop /mushr_mujoco_ros/buddy/control" />

    <!--launch-prefix="xterm -e gdb - -args" -->
    <node pkg="mushr_mujoco_ros" name="mushr_mujoco_ros" type="mushr_mujoco_ros_block_buddy" output="screen">
        <param name="mj_key" value="~/.mujoco/mjkey.txt" />
        <param name="model_file_path" value="$(find mushr_mujoco_ros)/models/$(arg environment).xml" />
        <param name="config_file_path" value="$(find mushr_mujoco_ros)/config/$(arg environment).yaml" />
        <param name="viz" value="true" />
    </node>

    <!-- Launch  map server-->
    <group if="$(arg map_server)">
        <include file="$(find mushr_sim)/launch/map_server.launch">
            <arg name="map" value="$(arg map_file)" />
        </include>
    </group>

    <!-- Launch  model/state publishers-->
    <param name="robot_description"
         textfile="$(find mushr_description)/robots/$(arg racecar_version).urdf"/>
</launch>
